source/server/listener_manager_impl.h

```diff
@@ -156,7 +169,7 @@ class ListenerImpl : public Listener,
   }
   Upstream::ClusterManager& clusterManager() override { return parent_.server_.clusterManager(); }
   Event::Dispatcher& dispatcher() override { return parent_.server_.dispatcher(); }
-  DrainManager& drainManager() override { return parent_.server_.drainManager(); }
+  Network::DrainDecision& drainManager() override { return *this; }
```

Comment:
This is a bit confusing since there is also a local drain manager. Could the member be renamed drainDecision()?